shared/src/simulation/behaviors/transfer-to-hospital.ts

```diff
@@ -0,0 +1,218 @@
+                        Object.keys(
+                            behaviorState.patientIdsSelectedForTransfer
+                        ).includes(patient.id)
+                    );
+                    const remainingPatients = difference(
+                        patients,
+                        selectedPatients
+                    );
+
+                    const groupedPatients = groupBy(patients, (patient) =>
+                        getVisiblePatientStatus(patient, draftState)
+                    );
+                    const groupedRemainingPatients = groupBy(
+                        remainingPatients,
+                        (patient) =>
+                            getVisiblePatientStatus(patient, draftState)
+                    );
+
+                    StrictObject.keys(patientStatusAllowedValues).forEach(
+                        (status) => {
+                            // If patients of this triage category just have been in this region
+                            if ((groupedPatients[status]?.length ?? 0) > 0) {
+                                // But there are no patients of this status right now
+                                if (
+                                    (groupedRemainingPatients[status]?.length ??
+                                        0) === 0
+                                ) {
+                                    // Then we want to report that we have finished this triage category
+                                    // Note:
+                                    // These conditions can be true for multiple times in one exercise if patient statuses change or new patients are added
+                                    addActivity(
+                                        simulatedRegion,
+                                        DelayEventActivityState.create(
+                                            nextUUID(draftState),
+                                            PatientCategoryTransferToHospitalFinishedEvent.create(
+                                                status,
+                                                true
+                                            ),
+                                            draftState.currentTime
+                                        )
+                                    );
+                                }
+                            }
+                        }
+                    );
+
+                    selectedPatients.forEach((patient) => {
+                        behaviorState.transferredPatientsCount[
+                            getVisiblePatientStatus(patient, draftState)
+                        ]++;
+                    });
+
+                    // The tick event itself is the last event per tick
+                    // Therefore, we can reset our state here
+                    behaviorState.patientIdsSelectedForTransfer = {};
+
+                    break;
+                }
```

Comment:
This pattern with temporary `patientIdsSelectedForTransfer` is really confusing and computationally expensive. How hard would it be to listen on `TransferPatientsInSpecificVehicleRequestEvent` or even `PatientTransferToHospitalSuccessfulEvent` and use this to catch the last patient instead?

Reply:
I have to keep `patientIdsSelectedForTransfer`, since otherwise, the behavior would try to load the same patient into multiple vehicles if more than one vehicle arrives in one tick, since there are some ticks between the selection and loading of a patient.

Therefore, I will also keep the calculation of the progress in the tick event handler, since I won't be able to remove `patientIdsSelectedForTransfer` anyways.